### PR TITLE
[assistant] - feature: filter global assistant based on provider

### LINKED ISSUE #5616: [Model Deactivation] hide deactivated providers global agents

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

Filter global assistants based on a provider whitelist before

## Risk

Moderated, we need to make sure the filters are correct

## Deploy Plan

Deploy `front`

### CODE REVIEW

**Review by spolu — CHANGES_REQUESTED** [review]
Let's go one step deeper and simply filter in `getGlobalAgents` in `global_agent.ts`.

1 line of code. Problem solved everywhere.

File: front/lib/api/assistant/global_agents.ts [hunk 1]
```diff
@@ -1046,10 +1047,15 @@ export async function getGlobalAgents(
...
 
+  const whiteListedProviders = owner.whiteListedProviders ?? MODEL_PROVIDER_IDS;
   const globalAgents: AgentConfigurationType[] = [];
 
   for (const agentFetcherResult of agentCandidates) {
-    if (agentFetcherResult) {
+    if (
+      agentFetcherResult &&
+      agentFetcherResult.scope === "global" &&
+      whiteListedProviders.includes(agentFetcherResult.model.providerId)
```
Comment (spolu) [inline front/lib/api/assistant/global_agents.ts:1057]: Please use the function we introduced to do that check

Comment (JulesBelveze) [reply front/lib/api/assistant/global_agents.ts]: Forgot about it, my bad

Comment (JulesBelveze) [reply front/lib/api/assistant/global_agents.ts]: Also added it in another file performing the same check

**Review by spolu — APPROVED** [review]

**Pull request merged**